### PR TITLE
Make Future tracking and stack traces optional

### PR DESCRIPTION
**Pull request by cheatfate**
- Make Future tracking optional via `-d:chronosFutureTracking` compilation flag.
- Make Future stack traces optional via `-d:chronosStackTraces` compilation flag.
- Address review comment https://github.com/status-im/nim-chronos/pull/107#pullrequestreview-442368768

### CODE REVIEW

File: chronos/asyncfutures2.nim [hunk 1]
```diff
@@ -36,8 +36,9 @@ type
     stackTrace: StackTrace ## For debugging purposes only.
     mustCancel*: bool
     id*: int
-    next*: FutureBase
-    prev*: FutureBase
+    when defined(chronosFutureTracking):
```
Comment (arnetheduck) [inline chronos/asyncfutures2.nim:39]: shouldn't this cover the stack trace as well? it's a pure debugging feature, and one of the most expensive things nbc does, according to profiling, when not syncing

Comment (cheatfate) [reply chronos/asyncfutures2.nim:39]: stack trace is not related to `chronosFutureTracking`.

Comment (arnetheduck) [reply chronos/asyncfutures2.nim:39]: yeah, but maybe it should be, naming this feature `chronosDebug` or something like that?

Comment (cheatfate) [reply chronos/asyncfutures2.nim:40]: `chronosStackTraces`.

**Pull request merged**